catkin_ws/src/state_estimation/src/sensors.py

```diff
@@ -19,111 +18,121 @@
 
 
 class Sensor():
-    def __init__(self, sensor_name):
-        self.time_before_considered_inactive = 1 #seconds
-        self.last_error_message_time = rospy.get_time()
+    def __init__(self, sensor_name, is_active=0):
+        self.is_active = is_active
         self.sensor_name = sensor_name
 
-        self.x = 0
-        self.y = 0
-        self.z = 0
-        
-        self.q_nwu_auv = np.quaternion(1, 0, 0, 0)
-        self.angular_velocity = np.array([0,0,0])
+    def get_sensor_name(self):
+        return self.sensor_name
 
-        # initialize a sensor as "inactive"
-        self.last_unique_state_time = -1 * self.time_before_considered_inactive
-        self.last_state = [self.x,self.y,self.z,self.q_nwu_auv,self.angular_velocity]
+    def sensor_status_cb(self, msg):
+        self.is_active = msg.data
+
```

Comment:
sensor_status_cb gets the message from the corresponding sensor's status topic and saves it to is_active var